dm/pkg/storage/utils.go

```diff
@@ -133,25 +135,36 @@ func CollectDirFiles(ctx context.Context, dir string, storage bstorage.ExternalS
 // RemoveAll remove files in dir.
 func RemoveAll(ctx context.Context, dir string, storage bstorage.ExternalStorage) error {
 	var err error
+	var backend *backuppb.StorageBackend
 	if storage == nil {
 		storage, err = CreateStorage(ctx, dir)
 		if err != nil {
 			return err
 		}
+		backend, err = bstorage.ParseBackend(dir, nil)
+		if err != nil {
+			return err
+		}
 	}
 
 	err = storage.WalkDir(ctx, &bstorage.WalkOption{}, func(filePath string, size int64) error {
 		err2 := storage.DeleteFile(ctx, filePath)
-		if errors.Cause(err2) == gstorage.ErrObjectNotExist {
-			// ignore not exist error when we delete files
+		// ignore not exist error when we delete files and backend is gcs/azure blob storage
+		if backend != nil && backend.GetGcs() != nil && errors.Cause(err2) == gstorage.ErrObjectNotExist {
+			return nil
+		}
+		if backend != nil && backend.GetAzureBlobStorage() != nil && bloberror.HasCode(err2, bloberror.BlobNotFound) {
 			return nil
 		}
 		return err2
 	})
 	if err == nil {
 		err = storage.DeleteFile(ctx, "")
-		if errors.Cause(err) == gstorage.ErrObjectNotExist {
-			// ignore not exist error when we delete files
+		// ignore not exist error when we delete files and backend is gcs/azure blob storage
+		if backend.GetGcs() != nil && errors.Cause(err) == gstorage.ErrObjectNotExist {
+			return nil
+		}
+		if backend.GetAzureBlobStorage() != nil && bloberror.HasCode(err, bloberror.BlobNotFound) {
```

Comment:
`backend` might be nil

Reply:
fix in https://github.com/pingcap/tiflow/pull/11913/commits/4560475723b25a613e4dd924c97db765d207e9e4